### PR TITLE
fix: get apollo-server-express working @3.x

### PR DESCRIPTION
**Pull request by astorm**
This PR adds support for the 3.x branch of `apollo-server-express`, and adjusts our tests to ensure apollo server starts correctly. 

Prior to this change, tests were failing with the message

```
TAP version 13
# POST /graphql
/Users/astorm/Documents/github/elastic/apm-agent-nodejs/node_modules/apollo-server-core/dist/ApolloServer.js:273
            throw new Error('You must `await server.start()` before calling `server.' +
            ^
Error: You must `await server.start()` before calling `server.applyMiddleware()`            
```

This PR adjusts the tests to start the server as expected in 3.x.

This PR also adjusts the `semver` constraint to allow version 3.x.

Finally, this PR adjusts the TAV tests to include an entry for apollo server 3.x using a peer dependency of `graphql@^15.0.0` and nodes >12.  These versions are based on 

https://github.com/apollographql/apollo-server/blob/94862a48ff9430d3238f9b456c9a99159fd0aa69/packages/apollo-server-express/package.json#L46

https://github.com/apollographql/apollo-server/blob/94862a48ff9430d3238f9b456c9a99159fd0aa69/packages/apollo-server-express/package.json#L26

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update documentation
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

File: lib/instrumentation/modules/apollo-server-core.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ const clone = require('shallow-clone-shim')
 module.exports = function (apolloServerCore, agent, { version, enabled }) {
   if (!enabled) return apolloServerCore
 
-  if (!semver.satisfies(version, '^2.0.2')) {
+  if (!semver.satisfies(version, '^2.0.2') && !semver.satisfies(version, '^3.0.0')) {
```
Comment (trentm) [inline lib/instrumentation/modules/apollo-server-core.js:10]: nit: could be done in a oner

```suggestion
  if (!semver.satisfies(version, '^2.0.2 || ^3.0.0')) {
```

File: test/instrumentation/modules/apollo-server-express.test.js [hunk 2]
```diff
@@ -14,6 +14,13 @@ var test = require('tape')
 var http = require('http')
 var express = require('express')
 var querystring = require('querystring')
+const semver = require('semver')
+// require('apollo-server-express') is a hard crash for nodes < 12.0.0
+const apolloServerExpressVersion = require('../../../node_modules/apollo-server-express/package.json').version
```
Comment (trentm) [inline test/instrumentation/modules/apollo-server-express.test.js:19]: can do this:

```suggestion
const apolloServerExpressVersion = require('apollo-server-express/package.json').version
```

File: test/config.test.js [hunk 3]
```diff
@@ -715,6 +715,9 @@ test('disableInstrumentations', function (t) {
   var expressGraphqlVersion = require('express-graphql/package.json').version
   var esVersion = require('@elastic/elasticsearch/package.json').version
 
+  // require('apollo-server-core') is a hard crash on nodes < 12.0.0
+  const apolloServerCoreVersion = require('../node_modules/apollo-server-core/package.json').version
```
Comment (trentm) [inline test/config.test.js:719]: can do this:

```suggestion
  const apolloServerCoreVersion = require('apollo-server-core/package.json').version
```

**Review by trentm — COMMENTED** [review]
So, other than the test tweaks for server start, the instrumentation just works? That's nice.

File: docs/supported-technologies.asciidoc [hunk 4]
```diff
@@ -86,7 +86,7 @@ The Node.js agent will automatically instrument the following modules to give yo
 A lot of higher level MongoDB modules use mongodb-core,
 so those should be supported as well.
 |https://www.npmjs.com/package/mongodb[mongodb] |>=2.0.0 <3.3.0 |Supported via mongodb-core
-|https://www.npmjs.com/package/mongodb[mongodb] |^3.3.0 <5 |Will instrument all queries
+|https://www.npmjs.com/package/mongodb[mongodb] |^3.3.0 |Will instrument all queries
```
[the remark this answers is not in the document]
Comment (astorm) [reply docs/supported-technologies.asciidoc:89]: look like it, fixed.

**Comment by astorm** [comment]
> So, other than the test tweaks for server start, the instrumentation just works? That's nice.

Correct -- whatever 2.x to 3.x changes were, they didn't impact our instrumentation.

**Review by trentm — APPROVED** [review]

**Pull request merged**